state-chain/primitives/src/lib.rs

```diff
@@ -234,7 +234,30 @@ pub struct SemVer {
 	pub minor: u8,
 	pub patch: u8,
 }
+
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub enum CfeCompatibility {
+	/// The version is currently compatible with the target.
+	Compatible,
+
+	/// The version is not yet compatible with the target. Should wait for the new version.
+	NotYetCompatible,
```

Comment:
And youll need to take BlockInfo instead of just the hash.

Reply:
> You could probably also merge change CreateBlockStreamError in be a sub-enum of a Incompatible variant of CfeCompatibility (i.e. have two variants instead of three).

Maybe I'm not fully understanding the suggestion, but I'm not convinced of this, we'd be leaking the CreateStateChainClientError into the SC? e.g. where we do `NodeCFEVersion::<T>::get(validator_id).compatibility_with_runtime(version) ==
						CfeCompatibility::Compatible` in the validator pallet. 

Also having it as a tuple is nice because the version doesn't have to be unpacked from every variant separately

Reply:
I see what you're saying, I'll have a look